docs/3_Building_Test_Data.md

```diff
@@ -21,11 +21,112 @@ You'll find resource about Test Data Builders
+## Live code the beginning of the Test Data-Builders here
+
+To help you to get started, here is a demo of how to use the Test Data-Builders in
+our particular situation.
+
+[![Video of the beginning of the solution](./images/Test_Data_Builders_YouTube_Video.png)](https://www.youtube.com/watch?v=TpOmhDZgEKc&feature=youtu.be)
+
+### Video Screen Shots
+
+#### Code Snippets
+
+<details>
+  <summary markdown='span'>
+  Code Snippets of NovelTestDataBuilder from the video
+  </summary>
+
+##### NovelTestDataBuilder.java 
+
+  ```java
+  package com.murex.tbw.domain.book;
+  
+  import com.google.common.collect.Lists;
+  import com.murex.tbw.domain.country.Language;
+  
+  public class NovelTestDataBuilder {
+      private double price = 10;
+  
+      public static NovelTestDataBuilder aNovel() {
+          return new NovelTestDataBuilder();
+      }
+  
+      public NovelTestDataBuilder costing(double price) {
+          this.price = price;
+          return this;
+      }
+  
+      public Novel build() {
+          return new Novel("Test Data Builders for Dummies", price, null, Language.ENGLISH, Lists.newArrayList());
+      }
+  }
+  ```
+</details>
+
+<details>
+  <summary markdown='span'>
+  Code Snippets of InvoiceTest from the video
+  </summary>
+
+##### InvoiceTest.java
+
+  ```java
+  package com.murex.tbw.purchase;
+  
+  import com.murex.tbw.domain.book.NovelTestDataBuilder;
+  import org.junit.jupiter.api.Assertions;
+  import org.junit.jupiter.api.Test;
+  
+  import static org.junit.jupiter.api.Assertions.*;
+
+  class InvoiceTest {
+      @Test
+      void Test_Data_Builders_Constraint_Applies_tax_rules_when_computing_total_amount() {
+          Invoice invoice = InvoiceTestDataBuilder.anInvoice()
+                  .from(USA)
+                  .with(PurchasedBookTestDataBuilder.aPurchasedBook()
+                          .of(NovelTestDataBuilder.aNovel()
+                                  .costing(50.0)
+                                  .build())
+                          .build())
+                  .build();
+          Assertions.assertEquals(50 * 1.15 * 0.98, invoice.computeTotalAmount());
+      }
+  }
+  ```
+
+</details>
+
+## DIY
 
 Now that you know everything about Test Data Builders, try to use them to
 improve your tests.
 
+### Delete the tests
+
+In real life, you would
+
+1. Fix the bugs
+2. Use test data-builders to instantiate your test data. 
+
+So let's Ignore or Comment-Out the previous tests!
+
+### Use Mikado Method to setup data for a test on Invoice
+
+We'll first focus on testing the [Invoice](../java/src/main/java/com/murex/tbw/purchase/Invoice.java) class.
+It's simpler and achievable given the time we have.
+
+To solve this section, use the test entitled *Test_Data_Builders_Constraint_Applies_tax_rules_when_computing_total_amount*!
+
+Use the test data-builder to instantiate your objects in the test.
+
+Make sure your project keeps building all the time. Notice every time you have something useful.
```

Comment:
I think we should not put this ~~'Make sure your project keeps building all the time'~~ Letting them go through code that does not build highlights a shortcoming of test data builders when used alone